tool/matomo/README.md

```diff
@@ -0,0 +1,22 @@
+# Matomo (formerly Piwik)
+- Set the Site ID
+- Choose whether you want image fallback tracking
+- Enter the URL to your Matomo install excluding http/https and trailing slashes
+- Choose whether you want to track admins (not recommended)
+- Choose whether you want to send Clean URLs (recommended): Matomo will aggregate Page Titles and show a nice waterfall cascade of all sites, 
+- Set alternative piwik.js URL for any purpose
+including categories and action types
+- Optional tracking for User ID
+- User ID could be id or username
+- Set the API url
+- Set the API token
+
+If both the API url and API token are set in the form and the siteid is empty when the form is submitted, an attempt to register the site with the API will be made.
+
+
+# Auto-provisioning
+- Set the global config settings 'apitoken' and 'apiurl' to enable auto provisioning. These can also be set in config.php e.g: 
+   - `$CFG->forced_plugin_settings['watool_matomo']['apiurl'] = 'https://matomo.org';`
+   - `$CFG->forced_plugin_settings['watool_matomo']['apitoken'] = 'xxxx';`
+- Auto provisioning attempts are made if the current site url has changed since any of the instances were stored.
```

Comment:
Can you please flesh this out a bit more, this might only need comments but might be some missing code too. I want to understand what is happened if we roll this auto provisioning out toa pre production side, eg p-client.catalyst-au.net and they get a matomo instance, and then a few months later they go live and the dns changes to moodle.client.com

In this scenario I can see a method called update_site but it isn't actually called from anywhere?

I would expect this to continue using the same matomo id, ie the one instance will seamlessly show stats from before and after the dns change. This should be the same as if the client later changes their domain for marketing reasons, the stats should continue to be seamless.

Reply:
It was previously just creating a new instance on the matomo server when the DNS changed but is now updating the url over the API if the DNS changes, essentially keeping the site id the same. Have fleshed this out in the docs.

Note that when manually adding an instance config on the front end it will also try to create/update over the API if the API config is in the form.

I have also documented these methods in unit tests for the watool_matomo plugin too.